includes/class-adst-plugins-stats-api.php

```diff
@@ -117,318 +83,553 @@ public function shortcode( $atts ) {
 		}
 
 		// Get the plugin data if it has already been stored as a transient.
-		$plugin_data = get_transient( 'bsf_tr_plugin_info_' . esc_attr( $atts['plugin'] ) );
+		$plugin_data = get_transient( 'new_bsf_tr_plugin_info_' . esc_attr( $plugin_slug ) );
 
 		// If there is no transient, get the plugin data from wp.org.
-		if ( ! $plugin_data ) {
-			$response = wp_remote_get( 'http://api.wordpress.org/plugins/info/1.0/' . esc_attr( $atts['plugin'] ) . '.json?fields=active_installs' );
+		if ( false === $plugin_data ) {
+			$response = wp_remote_get( 'http://api.wordpress.org/plugins/info/1.0/' . esc_attr( $plugin_slug ) . '.json?fields=active_installs' );
 
-			if ( is_wp_error( $response ) ) {
-				return 'Plugin slug is incorrect!';
-			} else {
-				$plugin_data = (array) json_decode( wp_remote_retrieve_body( $response ) );
+			$plugin_data = (array) json_decode( wp_remote_retrieve_body( $response ) );
```

Comment:
Sanitize plugin data array. Using the **array_map** function, you can define a callback function that will sanitize all elements from array.